spring-integration-core/src/test/java/org/springframework/integration/dsl/transformers/TransformerTests.java

```diff
@@ -287,6 +342,71 @@ public IdempotentReceiverInterceptor idempotentReceiverInterceptor() {
 			return idempotentReceiverInterceptor;
 		}
 
+		@Bean
+		public PollableChannel subFlowTestReplyChannel() {
+			return new QueueChannel();
+		}
+
+		@Bean
+		@Autowired
+		public IntegrationFlow syncSubFlowEnricher(SomeServiceGateway someServiceGateway) {
+			return f -> f
+					.enrich(e -> e.<TestPojo>requestPayload(p -> p.getPayload().getName())
+							.subFlow(sf -> sf
+									.<String>handle((p, h) -> someServiceGateway.deriveValueSync(p)))
+							.<String>headerFunction("foo", m -> m.getPayload())
+							.propertyFunction("name", m -> m.getPayload()))
+					.channel("subFlowTestReplyChannel");
+		}
+
+		@Bean
+		@Autowired
+		public IntegrationFlow asyncSubFlowEnricher(SomeServiceGateway someServiceGateway) {
+			return f -> f
+					.enrich(e -> e.<TestPojo>requestPayload(p -> p.getPayload().getName())
+							.subFlow(sf -> sf
+									.<String>handle((p, h) -> someServiceGateway.deriveValueAsync(p), ec -> ec.async(true)))
+							.<String>headerFunction("foo", m -> m.getPayload())
+							.propertyFunction("name", m -> m.getPayload()))
+					.channel("subFlowTestReplyChannel");
+		}
+
+		@Bean
+		public MessageChannel enricherReplyChannel() {
+			return MessageChannels.direct().get();
+		}
+
+		@Bean
+		@Autowired
+		public IntegrationFlow terminatingSubFlowEnricher(SomeService someService) {
+			return f -> f
+					.enrich(e -> e.<TestPojo>requestPayload(p -> p.getPayload().getName())
+							.subFlow(sf -> sf
+									.<String>handle(m -> someService.aTerminatingServiceMethod(m)))
+							.replyChannel("enricherReplyChannel")
+							.<String>headerFunction("foo", m -> m.getPayload())
+							.propertyFunction("name", m -> m.getPayload()))
+					.channel("subFlowTestReplyChannel");
+		}
+
+		@Bean
+		public GatewayProxyFactoryBean someServiceGateway() {
```

Comment:
I think you are aware of the `@MessagingGateway`  and `@IntegrationComponentScan`.
Maybe we really don't need this bean definition?